pkg/controller/datavolume-controller.go

```diff
@@ -225,7 +238,7 @@ func (c *DataVolumeController) Run(threadiness int, stopCh <-chan struct{}) erro
 
 	// Wait for the caches to be synced before starting workers
 	klog.V(2).Info("Waiting for informer caches to sync")
-	if ok := cache.WaitForCacheSync(stopCh, c.pvcsSynced, c.dataVolumesSynced); !ok {
+	if ok := cache.WaitForCacheSync(stopCh, c.pvcsSynced, c.dataVolumesSynced, c.snapshotClassSynced); !ok {
```

Comment:
Are we guaranteed to always have the snapshotClass available, so the sync can happen. We had an issue in the config-controller, where it was waiting for Routes in a k8s environment, and that would cause the controller to not initialize. I can see the same thing happening here in environments that don't have snapshots available. In the config-controller I solved it by checking for openshift and passing a slice to the WaitForCacheSync that had the RouteSync only if we are in openshift. Not sure if we can detect if the snapshots are available here.

Reply:
No, we are not guaranteed to have the snapshotClass... snapshot is a feature gate, and it's  an optional capability depending on the CSI plugin, and it's an external service.  You must try and reach the service and act accordingly based on the response.

Reply:
What do you by "have the snapshotClass available"?
We create the CRDs in createVolumeSnapshotClassCRD, though Michael commented that it is maybe not the right way.

Reply:
The Snapshotter endpoint may not even be on the cluster (Snapshot capability in the first place) and the plugin may or may not have snapshot capability at all.

Reply:
My main concern for this comment is, will WaitForCacheSync block forever waiting for a cache sync on a CRD that doesn't exist. I wasn't even worried about what John is talking about.

Reply:
We are currently creating the CRDs in 'factory.go -> createCRDResources', so it should guarantee we won't block on WaitForCacheSync. 
Regarding the snapshotClass availability, it should be validated in 'datavolume-controller -> getSnapshotClassForSmartClone' func. Do you think it's good enough for now?

Reply:
I still don't think we should create the CRDs.  And the controller (or operator) should handle the situation where they do not exist.  Maybe put the CRDs in a separate manifest that can be installed by users/CI but I feel pretty strongly that the operator should not install them.